trio/_core/_run.py

```diff
@@ -1048,8 +1130,15 @@ def aborted(raise_cancel):
                 # avoid a garbage cycle
                 # (see test_nursery_cancel_doesnt_create_cyclic_garbage)
                 del self._pending_excs
+        return None
 
-    def start_soon(self, async_fn, *args, name=None):
+    def start_soon(
+        self,
+        # TODO: TypeVarTuple
+        async_fn: Callable[..., Awaitable[object]],
+        *args: Any,
```

Comment:
I feel like `object` is fine here (unlike in the protocol, where I don't dare even think about variance xdd)